### DISCUSSION TITLE
Paste major version update dependency failure

### DISCUSSION THREAD

**Discussion by victoriahu** [twilio-labs/paste#1712] [General] [answered]
Hi team, 

We wanted to get some eyeballs on our latest Paste major version upgrade from 9.6.2 to 10.5.1 causing test failures, likely because of missing dependencies:
https://github.com/twilio/console/pull/233/checks?check_run_id=3186529664

We had an dependabot dependency auto-update and saw these test failures in the build.

147 out of 225 tests failed because they could not find module '@twilio-paste/theme', could not find module '@twilio-paste/box', or could not find module '@twilio-paste/styling-library'. @hharnisc and I were able to reproduce the test failures locally with the same 147 out of 225 tests failing as Dependabot. This seems to be a dependency issue with Paste. 

Would appreciate your help in troubleshooting this.

**Comment by SiTaggart** [comment]
Hi @victoriahu 

So it looks like this [PR](https://github.com/twilio/console/pull/226) recently updated the the lock file using npm v7, switching the `lockfileVersion` to 2.

So now, I believe when ever you upgrade Paste it is using NPM 7s new peer dependency logic and for what ever reason it just can't find any of the dependencies, at all. Very weird.

I have checked locally that if you update using npm v6, or with npm v7 but using `npm i --legacy-peer-deps` everything installs correctly, the paste deps can be found and things seem to run just fine.

I'm going to do a little more investigating because it does seem weird that npm v7 would being having so much trouble with Paste core and it's dependencies.

In the meantime we might have to set our supported node version to v14 so that npm v6 is used to install whilst we get it figured out.

**Comment by SiTaggart** [comment, marked as answer]
Hi @victoriahu 

From what I can tell there isn't anything wrong with our dependencies. I've created fresh Create React apps using NPM 7 and 6 and both work as expected, and running it finds all the dependencies.

What I suspect might be happening is that if you are using NPM 7, it's new way it handles peer dependencies is causing you issues potentially because you have some mis matched versions.

Right now, we state we support React ^16.8, and Console is running 17. You dependency on `react-hook-form-paste` also has a peer dependency requirement of Paste Core v8, and you're trying to install 10.

This probably doesn't cause you a problem when installing with a NPM 6 lock file as the legacy way it handles unmet peer dependency requirements irons out the issues. Because NPM 7 is more strict, I suspect the unmet peer dependency versions on React and Core are causing weird install bugs.

**Reply by victoriahu** [reply]
@SiTaggart We will downgrade to NPM 6 for now and delete and install a new lock file with NPM 6 for the meantime. Thank you so much!

**Reply by SiTaggart** [reply]
Let us know if that fixes it 🤞🏼
